docs: secure private access (#285)
aklivity/zilla-docs: src/solutions/concepts/kafka-proxies/secure-private-access.md

## Key Features

- Seamless MSK Serverless Connectivity across **Cross-Account VPCs**.
- **Custom Wildcard DNS** & Route 53 Hosted Zone Integration.
- **Unified Domain Name** for Kafka clients, streamlining configuration.
- **Eliminates** the need to **manually whitelist** each bootstrap endpoint to enable access.
- Seamless end-to-end **TLS** handshake.
- **Auto-Scaling** <ZillaPlus/> Instances.
- Deployed behind a **Network Load Balancer** for high availability and efficient request routing.

### Many-to-One Private Access

> Multiple Kafka clients from cross-account VPCs securely connect to a single Amazon MSK Serverless cluster. This approach simplifies multi-tenant access and ensures a unified, private connectivity model.

Reviewer: ```suggestion
Multiple Kafka clients from different cross-account VPCs securely connect to a single Amazon MSK Serverless cluster. This approach simplifies multi-tenant access and ensures a unified, private connectivity model.
```

![Many to One Private Access Overview](/many_to_one.png)

### One-to-Many Private Access

> Enables Kafka client to securely access multiple Amazon MSK Serverless clusters deployed across different VPCs.

Reviewer: ```suggestion
Enables Kafka clients to securely access multiple Amazon MSK Serverless clusters deployed across different VPCs.
```

![One to Many Private Access Overview](/one_to_many.png)

You will need to choose a wildcard DNS pattern to use for intranet access to the brokers in your Kafka cluster. These wildcard DNS names must resolve to the IP address of the VPC Endpoint in the client VPC, which then routes traffic via the VPC Endpoint Service to the ZillaPlus Network Load Balancer (NLB).

Additionally, the <ZillaPlus/> proxy must also be configured with a TLS server certificate representing the same wildcard DNS pattern.

## Deploy with CDK

> Follow the [Secure Private Access with CDK](https://github.com/aklivity/zilla-plus-aws-templates/tree/main/amazon-msk/cdk/secure-private-access) guide to generated or deploy a custom AWS CDK stack. This CDK-based deployment supports `SASL/SCRAM authentication`, `Mutual TLS (mTLS) authentication` or `Unauthorized access` to setup connectivity to your MSK Serverless cluster with a wildcard DNS pattern.

Reviewer: ```suggestion
Follow the [Secure Private Access with CDK](https://github.com/aklivity/zilla-plus-aws-templates/tree/main/amazon-msk/cdk/secure-private-access) guide to generate or deploy a custom AWS CDK stack.
```
Only IAM authorization is relevant for MSK Serverless clusters, there are no choices like MSK Provisioned clusters.
